google_guest_agent/main.go

```diff
@@ -262,21 +263,18 @@ func closer(c io.Closer) {
 }
 
 func main() {
-	ctx := context.Background()
-
-	var action string
-	if len(os.Args) < 2 {
-		action = "run"
-	} else {
-		action = os.Args[1]
-	}
-
-	if action == "noservice" {
-		runAgent(ctx)
-		os.Exit(0)
-	}
+	ctx, cancelContext := context.WithCancel(context.Background())
+	sigBus := make(chan os.Signal, 1)
+
+	signal.Notify(sigBus, syscall.SIGTERM, syscall.SIGINT)
+	go func() {
+		select {
```

Comment:
Since we don't expect multiple cases here we can simplify and log signal as - 
```
go func() {
        signal := <-sigBus
        // log agent is exiting because of signal it received
	cancelContext()
}()
```

Reply:
Indeed thanks.

Reply:
Done. Thanks.